src/org/traccar/database/DeviceManager.java

```diff
@@ -470,4 +507,13 @@ public void sendCommand(Command command) throws Exception {
         }
         return result;
     }
+
+    public void handleTextMessage(String phone, String message) {
```

Comment:
I think this method belongs to connection manager or something. Where do we throw most of the other events?